torch/csrc/jit/tensorexpr/expr.cpp

```diff
@@ -200,6 +200,10 @@ ExprHandle remainder(const ExprHandle& v1, const ExprHandle& v2) {
   return Intrinsics::make(kRemainder, v1, v2);
 }
 
+ExprHandle isnan(const ExprHandle& v1) {
+  return Intrinsics::make(kIsNan, v1);
```

Comment:
qq: What's the dtype of this `Intrinsics::make` expr?  Does it default to whatever the type of `v1` is?  That would be fine here, but I wonder if we should make our intrinsics typing better (cc @penguinwu who is interested in that :) ).

Reply:
It is by default v1, but I added logic so that if we see kIsNan we return torch.bool (which is what we need)